tests/unit/s2n_drbg_test.c

```diff
@@ -333,6 +333,11 @@ int check_drgb_version(s2n_drbg_mode mode, int (*generator)(void *, uint32_t), i
         }
 
         POSIX_GUARD_RESULT(s2n_drbg_wipe(&nist_drbg));
+        /**
+         * Can't rely on s2n_cleanup to clean s2n_init 
+         * becuase the default clean up callbacks are overwritten
+         * by s2n_rand_set_callbacks. */
```

Comment:
I think you can do better here. It's not clear why you're discussing s2n_init. How about:
```suggestion
        /* s2n_rand_set_callbacks overrode the default callbacks without cleaning up the default callbacks.
         * Reset the callbacks so that s2n_cleanup can later cleanup the original callbacks.
         */
```
Although reading that comment, I wonder if a clearer / less confusing fix would be to call s2n_rand_cleanup immediately before s2n_rand_set_callbacks. Would that cleanup the default callbacks and close the fd properly? 🤔 

Also, super nit: You're mixing block comment styles in a jarring way. You start with an extra empty comment line, but then end with no newline. Also, spelling.

Reply:
> It's not clear why you're discussing s2n_init. How about:

> Also, super nit: You're mixing block comment styles in a jarring way. You start with an extra empty comment line, but then end with no newline. Also, spelling.

I will use this comment and learn from this to write future comments.

> Would that cleanup the default callbacks and close the fd properly?

It will. I have implemented and tested it.

> I wonder if a clearer / less confusing fix would be to call s2n_rand_cleanup immediately before s2n_rand_set_callbacks. 

I am not sure if this implementation is clearer and less confusing than the current implementation. The current way is do rand cleanup after `s2n_drbg_wipe` which seems logical to me. To put `s2n_rand_cleanup` above `s2n_rand_set_callbacks` seems like calling a cleanup function out of the blue. We can discuss it offline if you disagree with me.

Reply:
> You have to cleanup the old callbacks before you set new ones.

This makes sense. I will do it in the next commit.